src/main/java/hudson/scm/SubversionSCM.java

```diff
@@ -2580,7 +2573,7 @@ private static boolean isAuthenticationFailedError(SVNCancelException e) {
 
     @CheckForNull
     private static DescriptorImpl descriptor() {
-        Jenkins instance = Jenkins.getInstance();
+        Jenkins instance = Jenkins.getInstanceOrNull();
```

Comment:
The key fix.

Reply:
Are we sure this code path is not reachable by the agent jvm?

Reply:
To the contrary, according to [JENKINS-34857](https://issues.jenkins-ci.org/browse/JENKINS-34857), it _is_ reachable from an agent JVM:

```
java.lang.IllegalStateException: Jenkins has not been started, or was already shut down
	at jenkins.model.Jenkins.getInstance(Jenkins.java:766)
	at hudson.scm.SubversionSCM.descriptor(SubversionSCM.java:2544)
	at hudson.scm.SubversionSCM.createDefaultSVNOptions(SubversionSCM.java:1043)
	at hudson.scm.SubversionSCM.createClientManager(SubversionSCM.java:1033)
	at hudson.scm.SubversionSCM$CheckOutTask.invoke(SubversionSCM.java:965)
	at hudson.scm.SubversionSCM$CheckOutTask.invoke(SubversionSCM.java:945)
	at …
	at ......remote call to VM177_Win7_64bits(Native Method)
	at …
	at hudson.FilePath.act(FilePath.java:996)
	at hudson.scm.SubversionSCM.checkout(SubversionSCM.java:894)
	at hudson.scm.SubversionSCM.checkout(SubversionSCM.java:830)
	at org.jenkinsci.plugins.multiplescms.MultiSCM.checkout(MultiSCM.java:143)
	at hudson.scm.SCM.checkout(SCM.java:485)
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1269)
	at …
```

which is exactly why I am switching to `getInstanceOrNull`—using `getInstance` here is illegal.

As noted in the other comment, it would of course be better to rewrite this code to have the `CheckOutTask` receive default SVN options as a constructor argument on the master side and have them be serialized in the normal way. That would be a more intrusive change, though.

Reply:
Reproducible for example from `PerJobCredentialStoreTest.remoteBuild` with patched core unless switched to `getInstanceOrNull`.

Reply:
LGTM